plugin/test/unit/templates/utils.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { tmpdir } from 'os'
+import { resolve, join, dirname } from 'path'
+
+import { copy, ensureDir, existsSync, readFileSync, unlink } from 'fs-extra'
+import { dir as getTmpDir } from 'tmp-promise'
+
+// eslint-disable-next-line import/no-namespace
+import * as templateUtils from '../../../src/templates/utils'
+
+const SAMPLE_PROJECT_DIR = `${__dirname}/../../../../demo`
+const TEST_TIMEOUT = 20_000
+
+/* eslint-disable node/prefer-global/process */
+const changeCwd = (cwd) => {
+  const originalCwd = process.cwd()
+  process.chdir(cwd)
+  return () => {
+    process.chdir(originalCwd)
+  }
+}
+
+// Move gatsby project from sample project to current directory
+const moveGatsbyDir = async () => {
+  await copy(SAMPLE_PROJECT_DIR, join(process.cwd()))
+}
+
+describe('prepareFilesystem', () => {
+  let cleanup
+  let restoreCwd
+  let downloadFileSpy
+
+  beforeEach(async () => {
+    const tmpDir = await getTmpDir({ unsafeCleanup: true })
+    restoreCwd = changeCwd(tmpDir.path)
+    // eslint-disable-next-line prefer-destructuring
+    cleanup = tmpDir.cleanup
+
+    downloadFileSpy = jest
+      .spyOn(templateUtils, 'downloadFile')
+      .mockResolvedValue()
+  })
+
+  afterEach(async () => {
+    // Cleans up the temporary directory from `getTmpDir()` and do not make it
+    // the current directory anymore
+    restoreCwd()
+    await cleanup()
+
+    jest.clearAllMocks()
+    jest.resetAllMocks()
+    jest.restoreAllMocks()
+    delete process.env.LOAD_GATSBY_LMDB_DATASTORE_FROM_CDN
+  })
+
+  it(
+    'downloads file from the CDN when LOAD_GATSBY_LMDB_DATASTORE_FROM_CDN is enabled',
+    async () => {
+      process.env.LOAD_GATSBY_LMDB_DATASTORE_FROM_CDN = 'true'
+      await moveGatsbyDir()
+
+      const cacheDir = resolve('.cache')
+      await templateUtils.prepareFilesystem(cacheDir)
+
+      expect(downloadFileSpy).toHaveBeenCalled()
+    },
+    TEST_TIMEOUT,
+  )
+
+  it(
+    'does not download file from the CDN when LOAD_GATSBY_LMDB_DATASTORE_FROM_CDN is not enabled',
+    async () => {
+      process.env.LOAD_GATSBY_LMDB_DATASTORE_FROM_CDN = 'false'
+      await moveGatsbyDir()
+
+      const cacheDir = resolve('.cache')
+      await templateUtils.prepareFilesystem(cacheDir)
+
+      expect(downloadFileSpy).not.toHaveBeenCalled()
+    },
+    TEST_TIMEOUT,
+  )
+
+  it(
+    'does not download file from the CDN when LOAD_GATSBY_LMDB_DATASTORE_FROM_CDN is undefined',
+    async () => {
+      await moveGatsbyDir()
+
+      const cacheDir = resolve('.cache')
+      await templateUtils.prepareFilesystem(cacheDir)
+
+      expect(downloadFileSpy).not.toHaveBeenCalled()
+    },
+    TEST_TIMEOUT,
+  )
+})
+/* eslint-enable node/prefer-global/process */
+
+describe('downloadFile', () => {
+  it('downloadFile can download a file', async () => {
```

Comment:
```suggestion
  it('can download a file', async () => {
```
[dust] The `downloadFile` bit is implied by the `describe` above.